modules/services/conky.nix

```
@@ -0,0 +1,55 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.services.conky;
+
+in with lib; {
+
+  meta.maintainers = [ maintainers.kaleo ];
+
+  options = {
+    services.conky = {
+      enable = mkEnableOption "conky, a light-weight system monitor";
+
+      package = mkPackageOption pkgs "conky" { };
+
+      settings = lib.mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Configuration written to {file}`$XDG_CONFIG_HOME/conky/conky.conf`.
+          Check <https://github.com/brndnmtthws/conky/wiki/Configurations"> for options.
+          If not set, the default configuration, as described by {command}`conky --print-config`, will be used.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions =
+      [ (hm.assertions.assertPlatform "services.conky" pkgs platforms.linux) ];
+
+    home.packages = [ cfg.package ];
+
+    xdg.configFile =
+      mkIf (cfg.settings != "") { "conky/conky.conf".text = cfg.settings; };
+
+    systemd.user.services.conky = {
+      Unit = {
+        Description = "Conky - Lightweight system monitor";
+        After = [ "graphical-session.target" ];
+      };
+
+      Service = {
+        Restart = "always";
+        RestartSec = "3";
+        ExecStart = toString ([ "${pkgs.conky}/bin/conky" ]
+          ++ optional (cfg.settings != "")
+          "--config ${config.xdg.configHome}/conky/conky.conf");
```

Is it necessary to place the configuration file in the user's home directory? Perhaps it would be possible to here do something like
```suggestion
          "--config ${pkgs.writeText "conky.conf" cfg.extraConfig}");
```
instead?

Then there is no need for the file in the home directory _and_ it makes it possible to auto-restart the service when the configuration changes (since the file path changes).

---

from tech perspective we don't have to put it in user's home directory. I did that as I saw lots of modules are doing that. e.g. https://github.com/nix-community/home-manager/blob/7e91f2a0ba4b62b88591279d54f741a13e36245b/modules/programs/rofi.nix#L219

but sounds those are kind of "legacy" way, I will follow your comment here to use the writeText instead.